### PR TITLE
Feature/en 2024

### PR DESCRIPTION
**Pull request by mikeweaver**

### CODE REVIEW

File: cielo24_PHP/src/cielo24/Actions.php [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+    {
+        $query_dict = $this->_initAccessReqDict($api_token);
+        $query_dict["api_securekey"] = $api_securekey;
+        // Nothing returned
+        WebUtils::httpRequest($this->BASE_URL, Actions::REMOVE_API_KEY_PATH, "GET", WebUtils::BASIC_TIMEOUT, $query_dict);
+    }
+
+    /// JOB CONTROL ///
+
+    /* Creates a new job. Returns an array of Guids where 'JobId' is the 0th element and 'TaskId' is the 1st element */
```
Comment (mikeweaver) [inline cielo24_PHP/src/cielo24/Actions.php]: Returning an array doesn't seem very user friendly. Can we return a dict or object instead?

File: cielo24_PHP/src/cielo24/Actions.php [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        $response = WebUtils::getJson($this->BASE_URL, Actions::DELETE_JOB_PATH, "GET", WebUtils::BASIC_TIMEOUT, $query_dict);
+        return $response["TaskId"];
+    }
+
+    /* Gets information about a job with job_id */
+    public function getJobInfo($api_token, $job_id)
+    {
+        $query_dict = $this->_initJobReqDict($api_token, $job_id);
+        $response = WebUtils::getJson($this->BASE_URL, Actions::GET_JOB_INFO_PATH, "GET", WebUtils::BASIC_TIMEOUT, $query_dict);
+        return $response;
```
Comment (mikeweaver) [inline cielo24_PHP/src/cielo24/Actions.php]: Returns the raw response? Shouldn't be make this more user friendly?

File: cielo24_PHP/src/cielo24/Enums.php [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    const STANDARD = "STANDARD";
+    const PRIORITY = "PRIORITY";
+    const CRITICAL = "CRITICAL";
+    const HIGH = "HIGH";
+}
+
+class Fidelity
+{
+    const MECHANICAL = "MECHANICAL";
+    const HIGH = "HIGH";
```
Comment (mikeweaver) [inline cielo24_PHP/src/cielo24/Enums.php]: HIGH and EXTERNAL are deprecated, do not include them. (Unless you need them to handle response data, in which case just add a comment indicating they are deprecated.)

File: cielo24_PHP/src/cielo24/Enums.php [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    const FAILED = "FAILED";
+}
+
+class Priority
+{
+    const ECONOMY = "ECONOMY";
+    const STANDARD = "STANDARD";
+    const PRIORITY = "PRIORITY";
+    const CRITICAL = "CRITICAL";
+    const HIGH = "HIGH";
```
Comment (mikeweaver) [inline cielo24_PHP/src/cielo24/Enums.php]: HIGH is deprecated. Please resolve this the same was as for Fidelity.

File: cielo24_PHP/src/cielo24/options/CommonOptions.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
+﻿<?php
+
+require_once("BaseOptions.php");
+
+/* Options found in both Transcript and Caption options
+ * All of the option properties are nullable. Properties that are null are ignored by the toQuery() method
+ * and are not part of the resulting query string.
+ */
+abstract class CommonOptions extends BaseOptions
```
Comment (mikeweaver) [inline cielo24_PHP/src/cielo24/options/CommonOptions.php]: Can we name this less generically? Perhaps commonformattingoptions?

File: cielo24_PHP/src/cielo24/Actions.php (hunk 1 above)
Comment (echistyakov) [inline cielo24_PHP/src/cielo24/Actions.php]: @mikeweaver WebUtils::getJson returns a dictionary

File: cielo24_PHP/src/cielo24/Actions.php (hunk 2 above)
Comment (echistyakov) [inline cielo24_PHP/src/cielo24/Actions.php]: @mikeweaver Similar to above, this is not a raw response, but a dictionary

File: cielo24_PHP/src/cielo24/Actions.php (hunk 1 above)
Comment (mikeweaver) [inline cielo24_PHP/src/cielo24/Actions.php]: @echistyakov The comment says that it will return an array. Which is correct?

Comment (echistyakov) [inline cielo24_PHP/src/cielo24/Actions.php]: @mikeweaver The comment is fixed now.

**Pull request merged**